core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -90,40 +90,65 @@ public void validateServiceRoles(Stack stack, InstanceGroupAdjustmentV4Request i
     public void validateClusterStatus(Stack stack) {
         Cluster cluster = stack.getCluster();
         if (cluster != null && !cluster.isAvailable()) {
-            throw new BadRequestException(format("Cluster '%s' is currently in '%s' state. Node count can only be updated if it's not available.",
+            throw new BadRequestException(format("Data Hub '%s' is currently in '%s' state. Node count can only be updated if it's not available.",
                     cluster.getName(), cluster.getStatus()));
         }
     }
 
     public void validateInstanceGroup(Stack stack, String instanceGroupName) {
         InstanceGroup instanceGroup = stack.getInstanceGroupByInstanceGroupName(instanceGroupName);
         if (instanceGroup == null) {
-            throw new BadRequestException(format("Stack '%s' does not have an instanceGroup named '%s'.", stack.getName(), instanceGroupName));
+            throw new BadRequestException(format("Data Hub '%s' does not have an group named '%s'.", stack.getName(), instanceGroupName));
         }
     }
 
+    public void validateScalabilityOfInstanceGroup(Stack stack, InstanceGroupAdjustmentV4Request instanceGroupAdjustmentJson) {
+        InstanceGroup instanceGroup = stack.getInstanceGroupByInstanceGroupName(instanceGroupAdjustmentJson.getInstanceGroup());
+        if (upscaleEvent(instanceGroupAdjustmentJson)) {
+            if (!instanceGroup.getScalabilityOption().upscalable()) {
+                throw new BadRequestException(format("Requested scaling up is forbidden on '%s' Data Hub '%s' group.",
+                        stack.getName(),
+                        instanceGroup.getGroupName()));
+            }
+        } else if (downScaleEvent(instanceGroupAdjustmentJson)) {
+            if (!instanceGroup.getScalabilityOption().downscalable()) {
+                throw new BadRequestException(format("Requested scaling down is forbidden on '%s' Data Hub '%s' group.",
```

Comment:
mayeb better: `Requested scaling down is forbidden on '%s' Data Hub's '%s' group.`